packages/ember-application/lib/system/application.js

```diff
@@ -256,28 +256,27 @@ var Application = Namespace.extend(DeferredMixin, {
   init: function() {
     this._super();
 
-    // Start off the number of deferrals at 1. This will be
-    // decremented by the Application's own `initialize` method.
-    this._readinessDeferrals = 1;
-
     if (!this.$) {
       this.$ = jQuery;
     }
 
-    // Create subclass of Ember.Router for this Application instance.
-    // This is to ensure that someone reopening `App.Router` does not
-    // tamper with the default `Ember.Router`.
-    // 2.0TODO: Can we move this into a globals-mode-only library?
-    this.Router = Router.extend();
     this.buildRegistry();
 
-    // TODO:(tomdale+wycats) Move to session creation phase
-    this.buildInstance();
-
     registerLibraries();
     logLibraryVersions();
 
-    this.scheduleInitialize();
+    // Start off the number of deferrals at 1. This will be
+    // decremented by the Application's own `initialize` method.
+    this._readinessDeferrals = 1;
+
+    if (true) {
+      // Create subclass of Ember.Router for this Application instance.
+      // This is to ensure that someone reopening `App.Router` does not
+      // tamper with the default `Ember.Router`.
+      // 2.0TODO: Can we move this into a globals-mode-only library?
+      this.Router = Router.extend();
+      this.waitForDOMReady(this.buildDefaultInstance());
```

Comment:
Yes, this is absolutely one of the goals of the refactor: to give developers fine-grained control over the boot process. While I would prefer to preserve the current DOM ready semantics, our goal is that developers can prevent the default autobooting behavior, and use the same API as the FastBoot harness to control how the app is booted.